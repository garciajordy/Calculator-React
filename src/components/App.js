import { useState } from 'react';
import ButtonPanel from './ButtonPanel';
import Display from './Display';
import calculator from '../logic/calculate';

function App() {
  const [value, setValue] = useState({
    total: null,
    next: null,
    operation: null,
  });
  const handleClick = (name) => {
    const data = { ...value };
    const result = calculator(data, name);
    setValue({ ...result });
  };
  const { total, next } = value;

  return (
    <>
      <Display result={(next && next.toString()) || (total && total.toString())} />
      <ButtonPanel handleClick={handleClick} />
    </>
  );
}

export default App;
