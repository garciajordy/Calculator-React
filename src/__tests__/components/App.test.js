import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import App from '../../components/App';

afterEach(cleanup);

it('should take a snapshot', () => {
  const { asFragment } = render(<App />);
  expect(asFragment(<App />)).toMatchSnapshot();
});
it('displays heading', () => {
  render(<App />);
  expect(screen.getByText('Ready to do some Maths?')).toBeInTheDocument();
});
