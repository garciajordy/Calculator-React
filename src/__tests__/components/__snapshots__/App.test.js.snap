// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Ready to do some Maths?
    </h1>
    <div
      class="class1"
    >
      <h1
        class="display"
      >
        0
      </h1>
      <div
        class="button-row"
      >
        <button
          class="button"
          type="button"
        >
          AC
        </button>
        <button
          class="button"
          type="button"
        >
          +/-
        </button>
        <button
          class="button"
          type="button"
        >
          %
        </button>
        <button
          class="button"
          style="background-color: orange;"
          type="button"
        >
          /
        </button>
      </div>
      <div
        class="button-row"
      >
        <button
          class="button"
          type="button"
        >
          7
        </button>
        <button
          class="button"
          type="button"
        >
          8
        </button>
        <button
          class="button"
          type="button"
        >
          9
        </button>
        <button
          class="button"
          style="background-color: orange;"
          type="button"
        >
          X
        </button>
      </div>
      <div
        class="button-row"
      >
        <button
          class="button"
          type="button"
        >
          4
        </button>
        <button
          class="button"
          type="button"
        >
          5
        </button>
        <button
          class="button"
          type="button"
        >
          6
        </button>
        <button
          class="button"
          style="background-color: orange;"
          type="button"
        >
          -
        </button>
      </div>
      <div
        class="button-row"
      >
        <button
          class="button"
          type="button"
        >
          1
        </button>
        <button
          class="button"
          type="button"
        >
          2
        </button>
        <button
          class="button"
          type="button"
        >
          3
        </button>
        <button
          class="button"
          style="background-color: orange;"
          type="button"
        >
          +
        </button>
      </div>
      <div
        class="button-row"
      >
        <button
          class="zero-btn"
          type="button"
        >
          0
        </button>
        <button
          class="button"
          type="button"
        >
          .
        </button>
        <button
          class="button"
          style="background-color: orange;"
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
